src/test/OneinchCallee.t.sol

```diff
@@ -0,0 +1,334 @@
+            hevm.store(token, keccak256(abi.encode(uint256(i), address(this))), bytes32(amount));
+            if (GemAbstract(token).balanceOf(address(this)) == amount) {
+                // Found it
+                return;
+            } else {
+                // Keep going after restoring the original value
+                hevm.store(token, keccak256(abi.encode(uint256(i), address(this))), prevValue);
+            }
+        }
+    }
+
+    address wethAddr;
+    address linkAddr;
+    address daiAddr;
+    address vatAddr;
+    address linkJoinAddr;
+    address daiJoinAddr;
+    address dogAddr;
+    address jugAddr;
+    address linkClipAddr;
+    address linkPipAddr;
+
+    Hevm hevm;
+    WethAbstract weth;
+    GemAbstract link;
+    VatAbstract vat;
+    DaiAbstract dai;
+    GemJoinAbstract linkJoin;
+    DogAbstract dog;
+    JugAbstract jug;
+    ClipAbstract linkClip;
+
+    OsmAbstract linkPip;
+
+    function setAddresses() private {
+        ChainlogHelper helper = new ChainlogHelper();
+        ChainlogAbstract chainLog = helper.ABSTRACT();
+        wethAddr = chainLog.getAddress('ETH');
+        linkAddr = chainLog.getAddress('LINK');
+        vatAddr = chainLog.getAddress('MCD_VAT');
+        daiAddr = chainLog.getAddress('MCD_DAI');
+        linkJoinAddr = chainLog.getAddress('MCD_JOIN_LINK_A');
+        daiJoinAddr = chainLog.getAddress('MCD_JOIN_DAI');
+        dogAddr = chainLog.getAddress('MCD_DOG');
+        jugAddr = chainLog.getAddress('MCD_JUG');
+        linkClipAddr = chainLog.getAddress('MCD_CLIP_LINK_A');
+        linkPipAddr = chainLog.getAddress('PIP_LINK');
+    }
+
+    function setInterfaces() private {
+        hevm = Hevm(hevmAddr);
+        weth = WethAbstract(wethAddr);
+        link = GemAbstract(linkAddr);
+        vat = VatAbstract(vatAddr);
+        dai = DaiAbstract(daiAddr);
+        linkJoin = GemJoinAbstract(linkJoinAddr);
+        dog = DogAbstract(dogAddr);
+        jug = JugAbstract(jugAddr);
+        linkClip = ClipAbstract(linkClipAddr);
+        linkPip = OsmAbstract(linkPipAddr);
+    }
+
+    VaultHolder ali;
+    address aliAddr;
+    OneinchCallee dan;
+    address danAddr;
+    address ONE_INCH_ROUTER;
+    bytes ONE_INCH_PARAMETERS;
+
+    function getPermissions() private {
+        hevm.store(dogAddr, keccak256(abi.encode(address(this), uint256(0))), bytes32(uint256(1)));
+        hevm.store(vatAddr, keccak256(abi.encode(address(this), uint256(0))), bytes32(uint256(1)));
+        hevm.store(linkPipAddr, keccak256(abi.encode(address(this), uint256(0))), bytes32(uint256(1)));
+        linkPip.kiss(address(this));
+    }
+
+    function setEnvVars() private {
+        // Execute this test with fresh data via
+        // `(cd scripts/1inch-callee-data-fetcher && npm ci && node index.js)`
+        // check the script for more details on how to use their API
+        uint256 ONE_INCH_BLOCK = hevm.envOr('ONE_INCH_BLOCK', 16327090);
+        hevm.rollFork(ONE_INCH_BLOCK);
```

Comment:
do you know what is the difference between rollFork and roll? (which at least I only used in the past). Also can you make sure this only affects the current test?

Reply:
> do you know what is the difference between rollFork and roll?

As far as I understood from the documentation, [`roll`](https://book.getfoundry.sh/cheatcodes/roll) will just set the block number to the one specified, without changing any state, while [`rollFork`](https://book.getfoundry.sh/cheatcodes/roll-fork) will also update state to the specified block. 

> Also can you make sure this only affects the current test?

Forge test [always reset the state between tests](https://github.com/foundry-rs/foundry/blob/b2baca32bd1a3b31b6f6ae2950a14c5bb8607cdb/forge/README.md?plain=1#L39-L40)

Reply:
makes sense